src/stan_math_backend/Lower_functions.ml

```diff
@@ -291,8 +286,9 @@ let collect_functors_functions (p : Program.Numbered.t) : defn list =
   let fun_decls, fun_defns =
     p.functions_block
     |> List.filter_map ~f:(fun d ->
-           if Set.mem fun_has_def d.fdname && Option.is_none d.fdbody then None
-           else Some (declare_and_define d) )
+           (* FIXME: external functions don't need decls
```

Comment:
Is this FIXME intended to stay in for the merge (I'm mostly fine if the answer is yes, just checking)

Reply:
Yes. It only affects external functions that are used directly in higher-order functions. You can't use math library functions like that either so I think this can wait until we have better support for external functions. It's not too difficult to fix once external functions are marked in some way in the MIR.